File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        nl={}
        for i in range(len(nums)):
            nl[nums[i]]=i
        for i in range(len(nums)):
            c = target - nums[i]
            if c in nl and nl[c]!=i:
                return [nl[c],i]
        return []

File: 1-two-sum/two-sum.go
func twoSum(nums []int, target int) []int {
    mp := make(map[int]int)
    for i,ele := range nums {
        if idx,found := mp[target-ele]; found {
            return []int{idx,i}
        }
        mp[ele] = i
    }
    return nil
}